library/alloc/src/string.rs

```diff
@@ -1246,11 +1240,11 @@ impl String {
         // SAFETY: start and end will be on utf8 byte boundaries per
         // the Searcher docs
         unsafe {
-            for (start, end) in matches {
-                ptr::copy(
-                    self.vec.as_mut_ptr().add(end - shrunk_by),
-                    self.vec.as_mut_ptr().add(start - shrunk_by),
-                    len - end,
+            for (start, end) in matches.into_iter().rev() {
+                let () = ptr::copy(
```

Comment:
Why is this `let ()` here?

Reply:
Style thing. This lets the reader know that there's no return value being discarded.

We use Rust pretty heavily in Fuchsia, and learned that all too often implicit drop can lead to `Result`s being dropped on the floor - so we always destructure on the LHS.

Reply:
Oh, I left the same comment. I think we should keep it similar to how rust handles it. Maybe fuchsia does but rust library `ptr::copy` does not do implicit drop. Also, `Result` have `must_use` and cannot be dropped like that.

Reply:
Reverted.